src/webapi/touch.rs

```diff
@@ -0,0 +1,138 @@
+    #[inline]
+    pub fn screen_x(&self) -> f64 {
+        js!(
+            return @{self.as_ref()}.screenX;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the Y coordinate of the touch point relative to the left edge of the screen.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Touch/screenY)
+    #[inline]
+    pub fn screen_y(&self) -> f64 {
+        js!(
+            return @{self.as_ref()}.screenY;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the X coordinate of the touch point relative to the left edge of the browser viewport, not including any scroll offset.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Touch/clientX)
+    #[inline]
+    pub fn client_x(&self) -> f64 {
+        js!(
+            return @{self.as_ref()}.clientX;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the Y coordinate of the touch point relative to the left edge of the browser viewport, not including any scroll offset.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Touch/clientY)
+    #[inline]
+    pub fn client_y(&self) -> f64 {
+        js!(
+            return @{self.as_ref()}.clientY;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the X coordinate of the touch point relative to the left edge of the document. Unlike clientX, this value includes the horizontal scroll offset, if any.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Touch/pageX)
+    #[inline]
+    pub fn page_x(&self) -> f64 {
+        js!(
+            return @{self.as_ref()}.pageX;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the Y coordinate of the touch point relative to the left edge of the document. Unlike clientX, this value includes the horizontal scroll offset, if any.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Touch/pageY)
+    #[inline]
+    pub fn page_y(&self) -> f64 {
+        js!(
+            return @{self.as_ref()}.pageY;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the Element on which the touch point started when it was first placed on the surface, even if the touch point has since moved outside the interactive area of that element or even been removed from the document.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Touch/target)
+    #[inline]
+    pub fn target(&self) -> Element {
+        js!(
+            return @{self.as_ref()}.target;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the X radius of the ellipse that most closely circumscribes the area of contact with the screen. The value is in pixels of the same scale as screenX.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Touch/radiusX)
+    #[inline]
+    pub fn radius_x(&self) -> f64 {
```

Comment:
Please leave a `TODO` here that this (and some of the others) should probably be changed to `f32` at a later date.

(Currently it doesn't matter as everything goes through the JS so it's ~turtles~ `f64` all the way down, but the specs say that those should be `f32` which I think will make a difference when we get native DOM bindings for WASM.)